src/app/app/layout.tsx

```diff
@@ -16,6 +18,16 @@ export default async function RootLayout({ children }: PropsWithChildren) {
 
   const loggedUser = await fetchUserWithSession(session);
 
+  const isUserOnboardingFinished = await user.isOboardingFinished(
+    loggedUser.username
+  );
+  const pathname = getOriginPath();
+  const isOnboardingPage = pathname.endsWith("onboarding");
+
+  if (!isUserOnboardingFinished && !isOnboardingPage)
+    redirect("/app/onboarding");
+  if (isUserOnboardingFinished && isOnboardingPage) redirect("/app");
```

Comment:
Not an issue, but eventually we should consider moving these redirect functions.

But no need to change, let's just keep an eye open if this starts to grow and become ugly.